### ISSUE TITLE
initialPosition .full Animation

### ISSUE THREAD

**Issue by lohenyumnam** [SCENEE/FloatingPanel#120]
when presenting the FloatingPanelController with initialPosition as full there the animation seems to wrongly presenting it as in the Image below 
![11](https://user-images.githubusercontent.com/17508663/51789968-2c725b80-21b5-11e9-94a4-fb46199fb416.gif)

but when I set the initialPosition to half the animation slide up which is the correct way.
![asdas](https://user-images.githubusercontent.com/17508663/51790004-9db20e80-21b5-11e9-9dcd-be565a934867.gif)

This method is use the show the panel

```Swift
func setupFloatingPanel() {
        fpc = FloatingPanelController()
        fpc.delegate = self
        let commentVC = CommentViewController()
        let commentNav = UINavigationController(rootViewController: commentVC)
        
        fpc.set(contentViewController: commentNav)
        fpc.isRemovalInteractionEnabled = true
        fpc.track(scrollView: commentVC.collectionView)
        fpc.contentInsetAdjustmentBehavior = .never
        
        // Initialize FloatingPanelController and add the view
        //        fpc.surfaceView.backgroundColor = .clear
        fpc.surfaceView.cornerRadius = 9.0
        fpc.surfaceView.shadowHidden = false
        self.present(fpc, animated: true)
    }
```

This is my custom FloatingPanelLayout  class

```Swift
class MyFloatingPanelLayout: FloatingPanelLayout {
    public var initialPosition: FloatingPanelPosition {
        return .half
    }
    
    public func insetFor(position: FloatingPanelPosition) -> CGFloat? {
        switch position {
        case .full: return 17.0 // A top inset from safe area
        case .half: return 316.0 // A bottom inset from the safe area
        case .tip: return 44.0 // A bottom inset from the safe area
        default: return nil // Or `case .hidden: return nil`
        }
    }
}
```

**Comment by SCENEE** [comment]
Could you let me know which version you use? It should be fixed at [this commit](https://github.com/SCENEE/FloatingPanel/commit/a095ace30e414df4cae97e179d49d75fce9d5401)  at v1.3.2.

**Comment by SCENEE** [comment]
Oh, I figure out the same issue when the instance is reused for presentation modally. I will try to fix! Thank you for letting me know!

**Comment by lohenyumnam** [comment]
@SCENEE One more thing with I dismiss the FloatingPanelController the ViewController, FloatingPanelController was presenting does not deallocated. deinit never called

**Comment by SCENEE** [comment]
I fixed the presentation animation problem at #122 👍 

On the other hand,  I can't confirm the above issue not calling deinit at v1.3.2. I try to repro it with the following code, but deinit is called when FloatingPanelController is dismissed.
```
diff --git a/Examples/Samples/Sources/ViewController.swift b/Examples/Samples/Sources/ViewController.swift
index f21a0d4..7b308cd 100644
--- a/Examples/Samples/Sources/ViewController.swift
+++ b/Examples/Samples/Sources/ViewController.swift
@@ -251,6 +251,7 @@ class SampleListViewController: UIViewController, UITableViewDataSource, UITable
             fpc.surfaceView.shadowHidden = false
 
             fpc.isRemovalInteractionEnabled = true
+            fpc.title = menu.name
 
             self.present(fpc, animated: true, completion: nil)
         default:
diff --git a/Framework/Sources/FloatingPanelController.swift b/Framework/Sources/FloatingPanelController.swift
index 95dfef7..a5d7980 100644
--- a/Framework/Sources/FloatingPanelController.swift
+++ b/Framework/Sources/FloatingPanelController.swift
@@ -161,6 +161,10 @@ public class FloatingPanelController: UIViewController, UIScrollViewDelegate, UI
                                       behavior: fetchBehavior(for: self.traitCollection))
     }
 
+    deinit {
+        print("<<<<<<<<<<<<<<<<<<<<< deinit >>>>>>>>>>>>>>>>", title)
+    }
+    
     private func didUpdateDelegate(){
         floatingPanel.layoutAdapter.layout = fetchLayout(for: traitCollection)
         floatingPanel.behavior = fetchBehavior(for: self.traitCollection)
```

**Comment by SCENEE** [comment]
Thanks! I merged it to the master.

> Any reason why this fix is not in the main branch ??

Because I'd like to check no breaking builds using CI 😄
